QuEST/src/CPU/QuEST_cpu.c

```diff
@@ -3871,3 +3871,57 @@ void agnostic_setDiagonalOpElems(DiagonalOp op, long long int startInd, qreal* r
         }
     }
 }
+
+int hasEvenParity(long long x, int* in_qubitIndices, int qs){
```

Comment:
- for a general number of qubits `qs`, this should use bitstrings, like [`getBitMaskParity`](https://quest-kit.github.io/QuEST-develop-doc/QuEST__cpu_8c_source.html#l03097). *However* this function appears to be only ever called on lists of length two, so should be using the exist [`isOddParity`](https://github.com/QuEST-Kit/QuEST/blob/a2b2a7d58a02515299d8db84ce233bcb0f38daca/QuEST/src/CPU/QuEST_cpu_internal.h#L38)

Otherwise, if this function needed to exist:
- should use existing convention of input names, e.g. `qubitInds` and `numQubits`
- this function appears to be called `op.numElemsPerChunk` * `numSumTerms` times, and hence should probably be inlined and defined in `QuEST_internal.h`. Better though, consideration of the algorithm it is invoked within may reveal it can be embedded more efficiently